### ISSUE TITLE
openHive(): read error: : Inappropriate ioctl for device

### ISSUE THREAD

**Issue by trobert-tsp** [minacle/chntpw#3]
Hello,

I used UUPDump to make an ISO of Windows Server 2025 Preview arm64 on macOS Sonoma 14.4.1. The version of chntpw I used is the latest (1.0.2) installed with `brew` from the `minacle/chntpw` tap.
I also tried with a version built from the latest source on this repo.

The zip package I downloaded is:
[26212.5000_arm64_en-us_multi_cb985455_convert.zip](https://github.com/minacle/chntpw/files/15332099/26212.5000_arm64_en-us_multi_cb985455_convert.zip)

Running the `uup_download_macos.sh` script led me to receive the following error:
```
Creating boot.wim...
openHive(): read error: : Inappropriate ioctl for device
chntpw: Unable to open/read a hive, exiting..
[WARNING] Ignoring extended attributes of 1 files
"ISODIR/sources/boot.wim" original size: 449132 KiB
Using LZX compression with 10 threads
Archiving file data: 1229 MiB of 1229 MiB (100%) done
"ISODIR/sources/boot.wim" optimized size: 441488 KiB
Space saved: 7644 KiB
```

Thank you if you can point me to any direction for a fix!

Best,
Thomas

**Comment by minacle** [comment]
I don’t know the specific cause, but the issue no longer occurred when I prevented chntpw from redirecting stdout to /dev/null.

While it may not be a perfect solution, I will provide a temporary workaround for now. Open the `uup_download_macos.sh` file in a text editor and modify it as follows:

```patch
 echo ""
 if [ -e ./files/convert.sh ]; then
   chmod +x ./files/convert.sh
+  sed -i '' '/chntpw -e /s/ >\/dev\/null//' ./files/convert.sh
   ./files/convert.sh wim "$destDir" 0
 fi
```

The `+` sign at the beginning of the line indicates that the line is newly added, and the actual content you need to add is this:

```sh
sed -i '' '/chntpw -e /s/ >\/dev\/null//' ./files/convert.sh
```

After doing this, run `uup_download_macos.sh`. Although the interactive messages that were originally suppressed by the script will now be displayed, the issue will no longer occur.

**Comment by trobert-tsp** [comment]
Thank you for the diff, I can confirm this workaround works.

Would you like me to close this issue as resolved or do you wish to investigate further?

Best,
Thomas

**Comment by minacle** [comment]
I would like to investigate this issue further, and I will close it myself once it is resolved.

**Comment by minacle** [comment]
This issue has been resolved in #4.
